packages/gasket-plugin-docusaurus/test/docs-view.test.js

```diff
@@ -0,0 +1,76 @@
+const assume = require('assume');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire');
+const fs = require('fs');
+const path = require('path');
+const testConfig = require('./docusaurus.test.config');
+
+const mockDocsConfigSet = { docsRoot: '/path/to/app/.docs' };
+const startStub = sinon.stub();
+const writeFileStub = sinon
+  .stub(fs.promises, 'writeFile')
+  .withArgs('/path/to/app/docusaurus.config.js', testConfig)
+  .resolves();
+
+const docsView = proxyquire('../lib/docs-view', {
+  '@docusaurus/core/lib': {
+    start: startStub
+  }
+});
+
+describe('docsView', () => {
+  let mockGasket;
+
+  beforeEach(() => {
+    mockGasket = {
+      exec: sinon.stub(),
+      metadata: {
+        app: {
+          name: 'App name'
+        }
+      },
+      config: {
+        root: '/path/to/app/'
+      }
+    };
+    sinon.resetHistory();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('reads docusaurus.config.js', () => {
+    const readConfigStub = sinon
+      .stub(fs.promises, 'readFile')
+      .withArgs('/path/to/app/docusaurus.config.js')
+      .resolves(testConfig);
+    const { root } = mockGasket.config;
+    assume(readConfigStub.withArgs(`${root}docusaurus.config.js`).returns(testConfig));
+  });
+
+  it('writes docusaurus.config.js if does not exist', () => {
+    const readConfigStub = sinon
+      .stub(fs.promises, 'readFile')
+      .withArgs('/path/to/app/docusaurus.config.js')
+      .rejects();
+    const { root } = mockGasket.config;
+    assume(readConfigStub.withArgs(`${root}docusaurus.config.js`).rejects());
+    assume(writeFileStub.withArgs(`${root}docusaurus.config.js`, testConfig).resolves());
+  });
```

Comment:
This test is only executing your stubs it seems. You want to execute the `docsView` handler _then_ check to see if calls to the writeFile are correct or not.